sdk/nodejs/output.ts

```diff
@@ -512,39 +567,36 @@ export function all<T1, T2, T3>(values: [Input<T1> | undefined, Input<T2> | unde
 export function all<T1, T2>(values: [Input<T1> | undefined, Input<T2> | undefined]): Output<[Unwrap<T1>, Unwrap<T2>]>;
 export function all<T>(ds: (Input<T> | undefined)[]): Output<Unwrap<T>[]>;
 export function all<T>(val: Input<T>[] | Record<string, Input<T>>): Output<any> {
-    if (val instanceof Array) {
-        const allOutputs = val.map(v => output(v));
-
-        const [syncResources, isKnown, isSecret, allResources] = getResourcesAndDetails(allOutputs);
-        const promisedArray = Promise.all(allOutputs.map(o => o.promise(/*withUnknowns*/ true)));
+    return output(val);
+}
 
-        return new Output<Unwrap<T>[]>(syncResources, promisedArray, isKnown, isSecret, allResources);
+function getAwaitableValue(v: any): any {
+    if (Output.isInstance(v)) {
+        return v.promise(/* withUnknowns */ true);
     } else {
-        const keysAndOutputs = Object.keys(val).map(key => ({ key, value: output(val[key]) }));
-        const allOutputs = keysAndOutputs.map(kvp => kvp.value);
-
-        const [syncResources, isKnown, isSecret, allResources] = getResourcesAndDetails(allOutputs);
-        const promisedObject = getPromisedObject(keysAndOutputs);
-
-        return new Output<Record<string, Unwrap<T>>>(syncResources, promisedObject, isKnown, isSecret, allResources);
+        return v;
     }
 }
```

Comment:
I have a hard time convincing myself that the above changes for `all` are functionally the same as the original code here.  How should I think about convincing myself this is a semantically equivalent implementation?  (other than trusting our tests)